modules/solid_properties/doc/content/modules/solid_properties/index.md

```diff
@@ -0,0 +1,185 @@
+
+Once your MOOSE object inherits from the `DerivativeMaterialInterface` class, you can now
+use material properties provided by the Solid Properties module. In the header file of the
+MOOSE object where a solid material property is needed (in this case, the
+thermal conductivity and its derivative), `const` references to the material properties are required:
+
+!listing modules/solid_properties/test/include/kernels/HeatDiffusion.h start=/// thermal end=};
+
+The names `_k` and `_dk_dT` are arbitrary, and may be selected as desired per application.
+
+The desired material properties are then obtained in the constructor in the source file
+according to the name of those properties defined in `ThermalSolidPropertiesMaterial` (which
+may be customized to be different from `k_solid`, `rho_solid`, and `cp_solid`). Here, the
+default name of `"k_solid"` is used.
+
+!listing modules/solid_properties/test/src/kernels/HeatDiffusion.C start=parameters end={
+
+Then, material properties are used just as other MOOSE materials are used. For example, the
+weak residual for the heat diffusion kernel grabs the value of thermal conductivity at the
+present quadrature point.
+
+!listing modules/solid_properties/test/src/kernels/HeatDiffusion.C start=computeQpResidual end=Real
+
+### Input file syntax
+
+The Solid Properties Materials are implemented in an input file in the `UserObjects` block.  For
+example, to use stainless steel 316 thermal properties to provide the thermal conductivity in
+the `HeatDiffusion` kernel above, the input file syntax would be:
+
+!listing modules/solid_properties/test/tests/stainless_steel_316/test.i
+  start=UserObjects
+  end=Kernels
+
+Due to the consistent interface for solid properties, a different solid can be substituted in the
+input file be changing the type of the userobject. For example, to set thermal properties
+with an arbitrary functional dependence instead, the solid property module section of
+the input file is:
+
+!listing modules/solid_properties/test/tests/functional/test.i
+  start=UserObjects
+  end=Kernels
+
+## Creating additional solids
+
+New solids can be added to the Solid Properties module by inheriting from the base class appropriate
+to the formulation and overriding the methods that describe the solid properties. New solids can also
+be added to separate applications for more custom usage. These can then be
+used in an identical manner as all other solid property userobjects.
+
+For example, suppose new thermal solid properties were desired for a material named `GorillaGlue` in
+the Navier-Stokes module. Begin from an empty material inheriting from `ThermalSolidProperties`.
+Override all methods in `ThermalSolidProperties` that you would like to imlement custom `GorillaGlue`
+properties for. The header file for your new material indicates all methods that will be defined.
+
+!listing language=cpp
```

Comment:
Does this work, so that you don't have to put the `//` lines?:
```
!listing! language=cpp
blah

blah
!listing-end!
```

Reply:
Yes, thanks for the suggestion!